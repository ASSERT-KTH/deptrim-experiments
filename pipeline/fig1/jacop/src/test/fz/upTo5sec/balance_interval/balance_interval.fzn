array [1..3] of int: X_INTRODUCED_11 = [1,-1,1];
var -5..5: bal:: is_defined_var;
var 1..5: X_INTRODUCED_8 ::var_is_introduced ;
var bool: X_INTRODUCED_22 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_23 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_37 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_38 ::var_is_introduced :: is_defined_var;
var 0..5: X_INTRODUCED_44 ::var_is_introduced ;
var 0..5: X_INTRODUCED_45 ::var_is_introduced ;
var 0..0: X_INTRODUCED_56 ::var_is_introduced  = 0;
array [1..5] of var int: x:: output_array([1..5]) = [6,4,3,3,4];
constraint int_lin_eq(X_INTRODUCED_11,[bal,X_INTRODUCED_45,X_INTRODUCED_8],0):: defines_var(bal);
constraint array_bool_or([X_INTRODUCED_23,X_INTRODUCED_38],true);
constraint int_max(4,0,X_INTRODUCED_44);
constraint int_max(1,X_INTRODUCED_44,X_INTRODUCED_45);
constraint int_eq_reif(X_INTRODUCED_8,0,X_INTRODUCED_22):: defines_var(X_INTRODUCED_22);
constraint array_bool_and([X_INTRODUCED_22],X_INTRODUCED_23):: defines_var(X_INTRODUCED_23);
constraint int_eq_reif(X_INTRODUCED_8,1,X_INTRODUCED_37):: defines_var(X_INTRODUCED_37);
constraint array_bool_and([X_INTRODUCED_37],X_INTRODUCED_38):: defines_var(X_INTRODUCED_38);
solve  satisfy;
