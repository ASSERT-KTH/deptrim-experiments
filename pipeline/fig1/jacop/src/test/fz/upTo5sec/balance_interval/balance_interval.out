x = array1d(1..5, [6, 4, 3, 3, 4]);
----------